Client connectivity issue

While running simulator I'm running into some exceptions when running with a client. Likely they are related to the io changed I made recently.

First exception:

```
7796, alive=false, type=JAVA_CLIENT] closed. Reason: Exception in NonBlockingSocketReader
java.lang.IllegalArgumentException: Client message frame length cannot be smaller than header size.
        at com.hazelcast.client.impl.protocol.ClientMessage.readFrom(ClientMessage.java:333)
        at com.hazelcast.client.impl.protocol.util.ClientMessageReadHandler.onRead(ClientMessageReadHandler.java:50)
        at com.hazelcast.nio.tcp.ClientReadHandler.onRead(ClientReadHandler.java:53)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingSocketReader.handle(NonBlockingSocketReader.java:163)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.handleSelectionKey(NonBlockingIOThread.java:349)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.handleSelectionKeys(NonBlockingIOThread.java:334)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.selectLoop(NonBlockingIOThread.java:252)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.run(NonBlockingIOThread.java:205)
```

I was hoping it would be the first packet running into this problem. I added some counters:

```
java.lang.IllegalStateException: Failed at packet: 20002
        at com.hazelcast.nio.tcp.ClientReadHandler.onRead(ClientReadHandler.java:58)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingSocketReader.handle(NonBlockingSocketReader.java:163)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.handleSelectionKey(NonBlockingIOThread.java:349)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.handleSelectionKeys(NonBlockingIOThread.java:334)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.selectLoop(NonBlockingIOThread.java:252)
        at com.hazelcast.nio.tcp.nonblocking.NonBlockingIOThread.run(NonBlockingIOThread.java:205)
Caused by: java.lang.IllegalArgumentException: Client message frame length cannot be smaller than header size.
        at com.hazelcast.client.impl.protocol.ClientMessage.readFrom(ClientMessage.java:333)
        at com.hazelcast.client.impl.protocol.util.ClientMessageReadHandler.onRead(ClientMessageReadHandler.java:50)
        at com.hazelcast.nio.tcp.ClientReadHandler.onRead(ClientReadHandler.java:56)
        ... 5 more
```

So it gets messed up somewhere in the stream.

--- a/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/ClientMessageReadHandler.java
+++ b/hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/ClientMessageReadHandler.java
@@ -50,6 +50,7 @@ public class ClientMessageReadHandler implements ReadHandler {
             final boolean complete = message.readFrom(src);
             if (!complete) {
                 messageCounter.inc(messagesCreated);
+                return;
             }
 
             //MESSAGE IS COMPLETE HERE

